Ktisis/Events/EventManager.cs

```diff
@@ -0,0 +1,36 @@
+﻿using FFXIVClientStructs.Havok;
+using Ktisis.Interface.Components;
+using Ktisis.Interface.Windows.ActorEdit;
+using Ktisis.Overlay;
+using Ktisis.Structs.Actor;
+using Ktisis.Structs.Bones;
+
+using System.Numerics;
+
+using static FFXIVClientStructs.Havok.hkaPose;
+
+namespace Ktisis.Events
+{
+    public static class EventManager
+    {
+        public unsafe delegate void TransformationMatrixChange(TransformTableState state, Matrix4x4 transformationMatrix, Bone? bone, Actor* actor);
+        public static TransformationMatrixChange? OnTransformationMatrixChange = null;
+
+        public delegate void GizmoChange(GizmoState state);
+        public static GizmoChange? OnGizmoChange = null;
+
+        public static unsafe void FireOnTransformationMatrixChangeEvent(TransformTableState state)
+        {
+            var bone = Skeleton.GetSelectedBone(EditActor.Target->Model->Skeleton);
+            var actor = (Actor*)Ktisis.GPoseTarget!.Address;
+            hkQsTransformf* boneTransform = 
+                bone is null ? &actor->Model->Transform : boneTransform = bone!.AccessModelSpace(PropagateOrNot.DontPropagate);
+            OnTransformationMatrixChange!(
```

Comment:
Perform a null-safe call, not a null-assertion call here:
```suggestion
            OnTransformationMatrixChange?(
```

Reply:
VS doesn't allow me to do that. Is that acceptable as an alternative?

```csharp
            if (OnTransformationMatrixChange is null) return;
            var bone = Skeleton.GetSelectedBone(EditActor.Target->Model->Skeleton);
	    var actor = (Actor*)Ktisis.GPoseTarget!.Address;
	    hkQsTransformf* boneTransform = bone is null ? &actor->Model->Transform : bone!.AccessModelSpace(PropagateOrNot.DontPropagate);
	    OnTransformationMatrixChange!(
		state,
		Interop.Alloc.GetMatrix(boneTransform), 
		Skeleton.GetSelectedBone(EditActor.Target->Model->Skeleton),
		actor
	    );

Reply:
That would be fine (and actually even better here), though I would prefer `OnTransformationMatrixChange == null` for the null-check.

For the other event listener calls, you can change to `?.Invoke(` (since there isn't anything expensive happening there).